### ISSUE TITLE
Error when running reload-certs.sh

### ISSUE THREAD

**Issue by nielscil** [bartowl/synology-stuff#2]
When trying to run your latest version of reload-certs.sh script I got these messages:

`jq: error: syntax error, unexpected IDENT, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
.13Reew.desc
jq: 1 compile error
jq: error: syntax error, unexpected IDENT, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
.13Reew.services|length
jq: 1 compile error`

Any idea how to fix this?

**Comment by crowbarz** [comment]
By pure coincidence, I also just encountered this bug and was here to propose a possible solution. The problem is that in both your case and mine `$domain_id` begins with a number, and jq doesn't like object identifiers that begin with a number. From https://stedolan.github.io/jq/manual/:

> Object Identifier-Index: .foo, .foo.bar
> [...]
> This syntax only works for simple, identifier-like keys, that is, keys that are all made of alphanumeric characters and underscore, and which do not start with a digit.
> 
> If the key contains special characters or starts with a digit, you need to surround it with double quotes like this: ."foo$", or else .["foo$"].

So I updated my copy of the script to quote `$domain_id` whenever it was used by jq, ie.
```
for domain_id in $(jq -r 'keys[]' $INFO); do
    domain=$(jq -r ".\"$domain_id\".desc" $INFO)
    num_services=$(jq -r ".\"$domain_id\".services|length" $INFO)
    src_path=$SYNO_CERT_ARCHIVE/$domain_id

#  echo "domain=$domain ($num_services):"
    for srv_id in $(seq 0 $((num_services - 1))); do
        name=$(jq -r ".\"$domain_id\".services[$srv_id].display_name" $INFO)
        service=$(jq -r ".\"$domain_id\".services[$srv_id].service" $INFO)
        subscriber=$(jq -r ".\"$domain_id\".services[$srv_id].subscriber" $INFO)
        isPkg=$(jq -r ".\"$domain_id\".services[$srv_id].isPkg" $INFO)
[...]
```
Note that you can't replace the double quotes around the jq script with single quotes as this would prevent `$domain_id` and `$srv_id` from being expanded as needed by the script.

@nielscil if these changes work for you as well, then I'll open a PR containing these updates.

**Comment by bartowl** [comment]
I will correct this to double quotes as it will fix the issue with domains beginning with a number, and won't break anything. Please make a PR and I'll merge it. Thanks.

**Comment by nielscil** [comment]
Sorry, I can't test the changes. I removed that certificate because I didn't use it anymore.

**Comment by bartowl** [comment]
As it looks to be exactly the case solved by #4 I will close this as solved. Thanks @crowbarz !
